/*  Copyright 2011 MaidSafe.net limited

    This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
    version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
    licence you accepted on initial access to the Software (the "Licences").

    By contributing code to the MaidSafe Software, or to this project generally, you agree to be
    bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
    directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
    available at: http://www.maidsafe.net/licenses

    Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
    under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
    OF ANY KIND, either express or implied.

    See the Licences for the specific language governing permissions and limitations relating to
    use of the MaidSafe Software.                                                                 */

option optimize_for = LITE_RUNTIME;

package maidsafe.drive.detail.protobuf;

message AttributesArchive {
  required uint64 st_size = 1;
  required uint64 creation_time = 2;
  required uint64 last_status_time = 3;
  required uint64 last_write_time = 4;
  required uint64 last_access_time = 5;
  required uint32 st_mode = 6;
  optional uint64 win_attributes = 7;
  optional bytes link_to = 8;
  optional uint32 st_dev = 9;
  optional uint32 st_nlink = 11;
  optional uint32 st_rdev = 14;
  optional uint32 st_blksize = 15;
  optional uint32 st_blocks = 16;
}

message MetaData {
  required bytes name = 1;
  required AttributesArchive attributes_archive = 2;
  optional bytes serialised_data_map = 3;
  optional bytes directory_id = 4;
}

message Directory {
  required bytes directory_id = 1;
  required uint32 max_versions = 2;
  repeated MetaData children = 3;
}
